src/pybind/mgr/cephadm/module.py

```diff
@@ -1665,15 +1665,49 @@ def _get_hosts(self, label=None):
                 r.append(h)
         return r
 
+    def _verify_add_host(self, spec: HostSpec):
+        """
+        Make sure, we have sane hostname.
+        """
```

Comment:
Maybe it's just me that took me a while to understand this function. Perhaps some comments would help.

So this results in cephadm either allowing all hosts as FQDN or as short hostnames? Is it that?

Wouldn't it be better to make this explicit? For example, FQDN would require '--fqdn' flag, otherwise short hostnames is assumed. I'm not much of a fan of validation rules changing implicitly.

Reply:
> Wouldn't it be better to make this explicit? For example, FQDN would require '--fqdn' flag, otherwise short hostnames is assumed. I'm not much of a fan of validation rules changing implicitly.

that would require 

https://github.com/ceph/ceph/blob/d347db28bcc04c1ee9266338549b5c5c8bd28b05/src/cephadm/cephadm#L2254

to be moved into ceph itself. might be something for a follow-up pr